components/connector/config/servers.go

```diff
@@ -45,7 +45,7 @@ func PrepareExternalGraphQLServer(cfg Config, certResolver api.CertificateResolv
 
 func PrepareHydratorServer(cfg Config, CSRSubjectConsts certificates.CSRSubjectConsts, externalSubjectConsts certificates.ExternalIssuerSubjectConsts, revokedCertsRepository revocation.RevokedCertificatesRepository, middlewares ...mux.MiddlewareFunc) (*http.Server, error) {
 	connectorCertHeaderParser := oathkeeper.NewHeaderParser(cfg.CertificateDataHeader, oathkeeper.ConnectorIssuer, oathkeeper.ConnectorCertificateSubjectMatcher(CSRSubjectConsts), cert.GetCommonName)
-	externalCertHeaderParser := oathkeeper.NewHeaderParser(cfg.CertificateDataHeader, oathkeeper.ExternalIssuer, oathkeeper.ExternalCertIssuerSubjectMatcher(externalSubjectConsts), cert.GetUUIDOrganizationalUnit)
+	externalCertHeaderParser := oathkeeper.NewHeaderParser(cfg.CertificateDataHeader, oathkeeper.ExternalIssuer, oathkeeper.ExternalCertIssuerSubjectMatcher(externalSubjectConsts), cert.GetUnmatchedOrganizationalUnit(externalSubjectConsts.OrganizationalUnitPattern))
```

Comment:
And we're going to remove the GUID from our regex in prod scenarios, right?

Reply:
I think that we also need to remove it from [charts/connector/values.yaml](https://github.com/kyma-incubator/compass/blob/708a616f4fbe85985b15d19e3c23fee5f82162fd/chart/compass/charts/connector/values.yaml#L23)